inspectit-oce-core/src/test/java/rocks/inspectit/oce/core/metrics/MeasuresAndViewsManagerTest.java

```diff
@@ -0,0 +1,243 @@
+
+            String metricName = "my-metric";
+            Measure existingMeasure = Measure.MeasureLong.create(metricName, "abc", "def");
+            View existingView = View.create(View.Name.create("existing"), "description",
+                    existingMeasure, Aggregation.LastValue.create(), Collections.emptyList());
+
+            MetricDefinitionSettings metricDefinition = MetricDefinitionSettings.builder()
+                    .unit("my-unit")
+                    .view("existing", ViewDefinitionSettings.builder().build())
+                    .view("newView", ViewDefinitionSettings.builder().build())
+                    .build()
+                    .getCopyWithDefaultsPopulated(metricName);
+
+            manager.addOrUpdateAndCacheMeasureWithViews(metricName, metricDefinition,
+                    Maps.newHashMap(metricName, existingMeasure), Maps.newHashMap("existing", existingView));
+
+            assertThat(manager.getMeasure(metricName)).contains(existingMeasure);
+
+            ArgumentCaptor<View> viewArg = ArgumentCaptor.forClass(View.class);
+            verify(viewManager, times(1)).registerView(viewArg.capture());
+            View view = viewArg.getValue();
+
+            assertThat(view.getMeasure()).isSameAs(existingMeasure);
+            assertThat(view.getName().asString()).isEqualTo("newView");
+            assertThat(view.getDescription()).contains(metricName);
+            assertThat(view.getAggregation()).isInstanceOf(Aggregation.LastValue.class);
+            assertThat(view.getColumns()).containsExactlyInAnyOrder(commonTags);
+        }
+
+        @Test
+        void testExceptionsDuringViewRegistrationHandled() {
+            doThrow(new RuntimeException()).when(viewManager).registerView(argThat(v -> v.getName().asString().equals("viewB")));
+            doNothing().when(viewManager).registerView(argThat(v -> !v.getName().asString().equals("viewB")));
+            String metricName = "my-metric";
+            MetricDefinitionSettings metricDefinition = MetricDefinitionSettings.builder()
+                    .unit("my-unit")
+                    .view("viewA", ViewDefinitionSettings.builder().build())
+                    .view("viewB", ViewDefinitionSettings.builder().build())
+                    .view("viewC", ViewDefinitionSettings.builder().build())
+                    .build()
+                    .getCopyWithDefaultsPopulated(metricName);
+
+            manager.addOrUpdateAndCacheMeasureWithViews(metricName, metricDefinition, emptyMap(), emptyMap());
+
+            ArgumentCaptor<View> viewArg = ArgumentCaptor.forClass(View.class);
+            verify(viewManager, times(3)).registerView(viewArg.capture());
+
+            List<String> names = viewArg.getAllValues().stream()
+                    .map(v -> v.getName().asString())
+                    .collect(Collectors.toList());
+            assertThat(names).containsExactlyInAnyOrder("viewA", "viewB", "viewC");
+
+        }
+    }
+
```

Comment:
Can you also some small test for `tryRecordingMeasurement` and verify that `updateMetricDefinitions` is not throwing any exception?